runtime/tests/decimal_conv.c

```diff
@@ -68,7 +68,38 @@ void testIntToDec() {
     validateIntToDec(INT64_MIN + 1, "-9223372036854775807");
 }
 
+void validateFloatToDec(double val, const char *decStr, PanicCode code) {
+    validate(_bal_decimal_from_float(val), decStr, code);
+}
+
+void testFloatToDec() {
+    validateFloatToDec(0.0, "0", 0);
+    validateFloatToDec(1.0, "1", 0);
+    validateFloatToDec(1.00000, "1", 0);
+    validateFloatToDec(1234567891.0, "1234567891", 0);
+    validateFloatToDec(1234567890123456.0, "1234567890123456", 0);
+    validateFloatToDec(1234567890123456.1, "1234567890123456", 0);
+    validateFloatToDec(1234567890123456.2, "1234567890123456.3", 0);
+    validateFloatToDec(1234567890123456.9, "1234567890123457", 0);
+    validateFloatToDec(1.7976931348623157e+308, "1.7976931348623157E+308", 0);
+    validateFloatToDec(1.7976931348623156e+308, "1.7976931348623155E+308", 0);
+    validateFloatToDec(1.7976931348623157e+307, "1.7976931348623158E+307", 0);
+    validateFloatToDec(1.7976931348623157e+308*10, "", PANIC_ARITHMETIC_OVERFLOW);
```

Comment:
Don't see the point of all these: we're not testing double. We just need to test double +ve/-ve infinity. So use something that is obviously +ve/-ve infinity 1.0/0.0, -1.0/0.0. Also need to test NaN.

Reply:
Changed